modules/services/gromit-mpx.nix

```
@@ -0,0 +1,212 @@
+        default = null;
+        example = 2;
+        description = ''
+          If not null, automatically draw an arrow at the end of a
+          stroke with the given size.
+        '';
+      };
+
+      modifiers = mkOption {
+        type = types.listOf (types.enum modsAndButtons);
+        default = [];
+        example = [ "SHIFT" ];
+        description = ''
+          Only activate this tool if the given modifiers are also active.
+        '';
+      };
+    };
+  };
+
+in
+{
+  meta.maintainers = [ maintainers.pjones ];
+
+  options.services.gromit-mpx = {
+    enable = mkEnableOption "Gromit-MPX annotation tool";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.gromit-mpx;
+      defaultText = "pkgs.gromit-mpx";
+      description = "The gromit-mpx package to use.";
+    };
+
+    hotKey = mkOption {
+      type = types.nullOr types.str;
+      default = "F9";
+      example = "Insert";
+      description = ''
+        A keysym that toggles the activate of gromit-mpx.  Set to
+        null to disable the hotkey in which case you'll have to
+        activate gromit-mpx manually using the command line.
+      '';
+    };
+
+    hotKeyCode = mkOption {
+      type = types.nullOr types.ints.positive;
+      default = null;
+      example = 118;
+      description = ''
+        A raw keycode that will toggle the activation of gromit-mpx.
+        Overrides the hotKey option.
+      '';
+    };
+
+    undoKey = mkOption {
+      type = types.nullOr types.str;
+      default = "F10";
+      description = ''
+        A keysym that causes gromit-mpx to undo the last stroke.  Use
+        this key along with the shift key to redo an undone stoke.
+        Set to null to disable the undo hotkey.
+      '';
+    };
+
+    undoKeyCode = mkOption {
+      type = types.nullOr types.ints.positive;
+      default = null;
+      example = 76;
+      description = ''
+        A raw keycode that causes gromit-mpx to undo the last stroke.
+        Overrides the undoKey option.
+      '';
+    };
+
+    opacity = mkOption {
+      type = types.addCheck types.float (f: f >= 0.0 && f <= 1.0);
+      default = 0.75;
+      example = 1.0;
+      description = "Opacity of the drawing overlay.";
+    };
+
+    tools = mkOption {
+      type = types.listOf (types.submodule toolOptions);
+      default = [
+        { device = "default"; type = "pen";    size=5; }
+        { device = "default"; type = "eraser"; size = 75; modifier = "3"; }
+      ];
+      defaultText = ''
+        [
+          { device = "default"; type = "pen";    size=5; }
+          { device = "default"; type = "eraser"; size = 75; modifier = "3"; }
+        ]
+      '';
+      description = ''
+        Tool definitions for gromit-mpx to use.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    xdg.configFile."gromit-mpx.cfg".text =
+      concatStringsSep "\n"
+        (zipListsWith toolToCfg (range 1 (length cfg.tools)) cfg.tools);
```

I believe this is the same as
```suggestion
        (imap1 toolToCfg cfg.tools);
```
using `imap1` from the `lists.nix` library in Nixpkgs.